File: src/ingestion/adapters/archive_adapter.ts
import { AdapterResult } from './html_adapter';

export class ArchiveAdapter {
  name = 'archive';

  async extract(input: string): Promise<AdapterResult> {
    return {
      content: input,
      contentType: 'application/archive',
      metadata: { adapter: this.name }
    };
  }
}

File: src/ingestion/adapters/archive_adapter.py
"""Archive fallback adapter.

Used when primary official endpoints are temporarily unavailable.
"""

from .base_adapter import SourceAdapter


class ArchiveAdapter(SourceAdapter):
    name = "archive_adapter"

    async def fetch(self, url: str):
        return {
            "adapter": self.name,
            "url": url,
            "status": "fallback_required",
        }
